src/styles/blocks/calendar.scss

```
@@ -0,0 +1,60 @@
+.calendar {
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  width: 100vw;
+  height: 100vh;
+
+  padding: 0 10px;
+
+  @each $length, $value in $monthLength {
+    @for $index from $value through 32 {
+      &--month-length--#{$length} {
+        .calendar__square:nth-child(#{$index}) {
+          display: none;
+        }
+      }
+    }
+  }
+
+  @each $day, $value in $startDay {
+    &--start-day--#{$day} {
+      .calendar__square:first-child {
+        margin-left: startMonth($value, $squareSize, $calendarGap);
+      }
+    }
+  }
+
+  &__container {
+    display: flex;
+    flex-wrap: wrap;
+    gap: 1px;
+    width: calendarWidth($squareSize, $calendarGap, $calendarColumns);
+  }
+
+  &__square {
+    width: $squareSize;
+    height: $squareSize;
+    background-color: #eee;
+    border: 1px solid black;
+
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    text-align: center;
+
+    cursor: pointer;
+    transition-duration: 0.5s;
```

Always list styles that you apply transition for:
```suggestion
    transition: background-color 0.5s, transform 0.5s;
```